hyperon_das/query_engines.py

```diff
@@ -255,6 +271,30 @@ def get_links(
         if not local:
             return self.remote_das.get_links(link_type, target_types, link_targets)
 
+    def get_incoming_links(
+        self, atom_handle: str, **kwargs
+    ) -> List[Union[Tuple[Dict[str, Any], List[Dict[str, Any]]], Dict[str, Any]]]:
+        local_links = self.local_query_engine.get_incoming_links(atom_handle, **kwargs)
+        remote_links = self.remote_das.get_incoming_links(atom_handle, **kwargs)
+
+        if kwargs.get('handles_only', False):
+            local_links_set = set(local_links)
+            remote_links_set = set(remote_links)
+            return list(local_links_set.union(remote_links_set))
+        else:
+            remote_links_set = {link['handle']: link for link in remote_links}
```

Comment:
The name is misleading since this is not a `set`. Either rename it to `remote_links_dict` or find a better name.